cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/service/BusinessMetadataStore.java

```diff
@@ -48,11 +56,17 @@
 
   private static final Id.DatasetInstance BUSINESS_METADATA_INSTANCE_ID =
     Id.DatasetInstance.from(Id.Namespace.SYSTEM, "business.metadata");
+  private static final Map<String, String> EMPTY_PROPERTIES = ImmutableMap.of();
+  private static final Set<String> EMPTY_TAGS = ImmutableSet.of();
+
   private final Transactional<BusinessMdsIterable, BusinessMetadataDataset> txnl;
+  private final boolean publishEnabled;
+  private final MetadataChangePublisher changePublisher;
 
   @Inject
-  public BusinessMetadataStore(TransactionExecutorFactory txExecutorFactory,
-                               @Named(DataSetsModules.BASIC_DATASET_FRAMEWORK) final DatasetFramework dsFramework) {
+  BusinessMetadataStore(TransactionExecutorFactory txExecutorFactory,
+                        @Named(DataSetsModules.BASIC_DATASET_FRAMEWORK) final DatasetFramework dsFramework,
```

Comment:
JIRA for refactoring Guice bindings https://issues.cask.co/browse/CDAP-3661?focusedCommentId=18490&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-18490